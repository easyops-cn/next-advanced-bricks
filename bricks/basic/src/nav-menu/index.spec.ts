import { describe, test, expect } from "@jest/globals";
import { act } from "react-dom/test-utils";
import "./index.js";
import { NavMenu } from "./index.js";

jest.mock("@next-core/theme", () => ({}));

jest.mock("@next-core/runtime", () => ({
  getHistory: () => ({
    location: {},
    listen: () => {
      //
    },
    createHref: () => "/oops",
  }),
  matchPath: () => true,
}));

describe("basic.nav-menu", () => {
  test("basic usage", async () => {
    const element = document.createElement("basic.nav-menu") as NavMenu;

    expect(element.shadowRoot).toBeFalsy();
    act(() => {
      element.menu = {
        title: "mock a",
        menuItems: [
          {
            text: "菜单一",
            type: "default",
          },
          {
            title: "菜单二",
            type: "subMenu",
            items: [
              {
                text: "菜单二-1",
                type: "default",
              },
              {
                title: "分组",
                type: "group",
                items: [
                  {
                    text: "分组1",
                    to: "/oops",
                    type: "default",
                  },
                  {
                    text: "分组2",
                    type: "default",
                  },
                ],
              },
            ],
          },
        ],
      };
      document.body.appendChild(element);
    });
    expect(element.shadowRoot).toBeTruthy();
    expect(element.shadowRoot?.childNodes.length).toBe(2);

    expect(element.shadowRoot?.innerHTML).toMatchSnapshot();

    act(() => {
      document.body.removeChild(element);
    });

    expect(document.body.contains(element)).toBeFalsy();
  });
});
