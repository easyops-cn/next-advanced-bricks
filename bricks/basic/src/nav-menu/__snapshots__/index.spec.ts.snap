// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic.nav-menu basic usage 1`] = `"<style>nav-menu.shadow.css</style><div class="nav-menu-wrapper"><basic.general-menu-item title=""><basic.general-link><span class="menu-item-label">菜单一</span></basic.general-link></basic.general-menu-item><basic.general-popover class="popover" trigger="hover" placement="bottom-start"><basic.general-menu-item class="sub-menu-item-label" slot="anchor" title=""><span class="menu-item-label">菜单二</span></basic.general-menu-item><div class="sub-menu-wrapper"><basic.general-menu-item title=""><basic.general-link><span class="menu-item-label">菜单二-1</span></basic.general-link></basic.general-menu-item><div class="group-label">分组</div><div class="group-wrapper"><basic.general-menu-item title="" active=""><basic.general-link url="/oops"><span class="menu-item-label">分组1</span></basic.general-link></basic.general-menu-item><basic.general-menu-item title=""><basic.general-link><span class="menu-item-label">分组2</span></basic.general-link></basic.general-menu-item></div></div></basic.general-popover></div>"`;
